### DISCUSSION TITLE
Troops arrangement during battle with neutral monsters

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#4478] [General]
I'm not sure, I've faced all the cases, that's why I create a discussion. I hope other people would share other cases, if I missed something.

There are different neutral monsters' arrangement style in the OG, comparing to fheroes2:
![image](https://user-images.githubusercontent.com/55348946/138558917-e1a405a5-578d-4635-9fbe-378442c8d67d.png)
![image](https://user-images.githubusercontent.com/55348946/138558925-97d6c09c-eef5-4c7f-bbd6-e3ab8845c07f.png)
![image](https://user-images.githubusercontent.com/55348946/138558957-0d134d09-b11c-4227-acdb-37873205eb59.png)
![image](https://user-images.githubusercontent.com/55348946/138558991-b2221841-3cb9-49fb-9fb5-0bc654b27b2e.png)
![image](https://user-images.githubusercontent.com/55348946/138559233-d63f7a81-1536-4ebd-8941-0985e500d4de.png)
![image](https://user-images.githubusercontent.com/55348946/138559289-e6bc9612-e4c4-4895-ada0-f4ccf481b27a.png)
![image](https://user-images.githubusercontent.com/55348946/138559299-0ea9acbe-3019-4c74-8a3e-0fa846659a23.png)
![image](https://user-images.githubusercontent.com/55348946/138559328-b9b7c0f3-1ecf-4c1e-8885-ba72eae42cbb.png)
![image](https://user-images.githubusercontent.com/55348946/138559337-1fd1f4fd-fa90-442a-bd45-d73aaa407c4c.png)
![image](https://user-images.githubusercontent.com/55348946/138559365-64260a67-4703-4986-949d-d83c1c5c6704.png)
![image](https://user-images.githubusercontent.com/55348946/138559394-937643e0-5ea9-487d-8efa-1227facb3c92.png)
![image](https://user-images.githubusercontent.com/55348946/138559427-3d853194-89c6-487e-8927-7a0ea4ad56f9.png)
![image](https://user-images.githubusercontent.com/55348946/138559503-ee54951e-1524-4523-a096-915ee5da2849.png)
![image](https://user-images.githubusercontent.com/55348946/138560495-d2650b20-2382-4c87-86d8-bd062f5c66bc.png)
![image](https://user-images.githubusercontent.com/55348946/138560499-937ae776-3f80-4e3a-ae9c-13505a2991d0.png)
![image](https://user-images.githubusercontent.com/55348946/138560536-35cfdda9-4b22-4706-b900-23b017f9758b.png)

I've tried a couple of battles in the OG and what I've noticed.

Ranged troops are always arranged to as many stacks, as possible.

Also melee troops can sometimes miss higher slot and place 3 stacks in center  (It happens really rare), while ranged are always arranged starting from upper slot.
![image](https://user-images.githubusercontent.com/55348946/138560374-275ab55a-f3db-487f-9a1f-223705b1270d.png)
![image](https://user-images.githubusercontent.com/55348946/138560396-fc32e63c-4788-451b-91a7-801dd200d176.png)
Upgraded stack is always placed in central position.
While such troops as melee Giants (melee + ranged stack), as on the screenshots, could be arranged sometimes into 2 stacks  

So what's the conclusion?
Troops could never be placed in one stack (unless there's only unit) (Which is the case of fheroes2)
3 stacks can be arranged starting from the higher slot.
The most of time there appear 5 stacks on a battlefield. (4, 3 or 2 are rare cases when the army is not so big)

**Comment by Branikolog** [comment]
I'm not sure, we should copy everything here from the OG, but We definitely missing some variety: currently in fheroes2 the most common cases are 3 or 1 stacks., while in the OG there're 5.

**Comment by LeHerosInconnu** [comment]
Yes, there is more variety in the original game, and it also depends on the number of troops of the attacker, but I don't know the exact rules.

Also, in fheroes2, when the number of creatures of the same type (all upgraded or all not upgraded) is not evenly distributed, the troop in the center has a larger number of creatures (probably for the purpose of graphical symmetry) but this is not the best way to position the troops from a tactical point of view because it is the troop positioned at the top of the screen that plays first.

In fheroes2:

![Creatures number 001](https://user-images.githubusercontent.com/43583994/139309571-3b6ddd81-bef5-4f6a-881e-b8ee26a74258.png)

![Creatures number 002](https://user-images.githubusercontent.com/43583994/139309592-d7d07970-84ba-47be-8977-8a9ecad3c7e6.png)

![Creatures number 003](https://user-images.githubusercontent.com/43583994/139309610-ac86d758-9c72-4f57-bc57-93e9f806d6eb.png)

**Comment by ihhub** [comment]
Ah. I wanted to add a smart monster positioning before the battle as a part of `God's level` of difficulty... For example you're attacking 10 Phoenixes with slower but stronger army. Phoenixes can be divided into 2 groups: 1 and 9. 1 hits your stack and then 9 does the same without having retaliation. We can argue about this but it's just an idea for now.

I agree that we shouldn't copy the placement of neutral monsters from the original game to be less predictable for players and have more enjoyable battles. We must add logic to distribute monsters based on the army of a player. For now we can define rules to do this. They can be simple for now but at least better than current implementation.

**Reply by Branikolog** [reply]
Is it correct to change distribution based on hero's army? I mean two heroes with different armies would face different enemy arrangement?
